appmap/test/test_http.py

```diff
@@ -0,0 +1,32 @@
+# pylint: disable=missing-function-docstring,missing-function-docstring,missing-module-docstring
+# pylint: disable=redefined-outer-name,unused-argument,unused-import
+import httpretty
+import pytest
+import requests
+
+import appmap.http
+
+def test_http_client_capture(mock_requests, events):
+    requests.get('https://example.test/foo/bar?q=one&q=two&q2=%F0%9F%A6%A0')
+
+    request = events[0]
+    assert request.http_client_request == {
+        'request_method': 'GET',
+        'url': 'https://example.test/foo/bar',
+        'headers': { 'Connection': 'keep-alive' },
+    }
+    message = request.message
+    assert message[0].items() >= { 'name': 'q', 'value': "['one', 'two']" }.items()
+    assert (message[1].items() >= { 'name': 'q2', 'value': "'🦠'" }.items()) \
+        or (message[1].items() >= { 'name': 'q2', 'value': "'\\U0001f9a0'" }.items())
+
+    assert events[1].http_client_response.items() >= {
+        'status_code': 200,
+        'mime_type': 'text/plain; charset=utf-8'
+    }.items()
+
+@pytest.yield_fixture
+def mock_requests():
```

Comment:
Apparently, this can just be `@pytest.fixture` now: https://docs.pytest.org/en/6.2.x/yieldfixture.html .

Also, you might want to use the `name` argument to `fixture`, and get rid of `pylint: disable=redefined-outer-name`: https://docs.pytest.org/en/6.2.x/reference.html#pytest-fixture .